Add Testing for Cross-Device Session Reuse and Single Authentication Cookie in SSO Environments [WSTG-SESS-06] (#1379)
OWASP/wstg: document/4-Web_Application_Security_Testing/06-Session_Management_Testing/06-Testing_for_Logout_Functionality.md

> It is expected that the invocation of a log out function in a web application connected to a SSO system or in the SSO system itself causes global termination of all sessions. An authentication of the user should be required to gain access to the application after log out in the SSO system and connected application.

> ### Testing for Cross-Device Session Reuse and Single Authentication Cookie in Single Sign-On Environments

Reviewer: ```suggestion
### Testing Cross-Device Session Reuse and Single-Value Dependency in SSO
```

In some Single Sign-On (SSO) implementations, authentication across multiple applications relies on a single authentication cookie issued after successful login (e.g., after password and MFA verification).

While this cookie may be marked Secure and HttpOnly, its security depends entirely on server-side validation and revocation controls. If the SSO ecosystem relies on a single cookie as the sole proof of authentication, compromise of that cookie may grant immediate cross-application access.

Testers should evaluate whether:

- Only one cookie is required to maintain authenticated state.
- The cookie can be reused across browsers or devices.
- Logout properly invalidates the cookie server-side.
- All applications trusting the SSO cookies reject it after logout.

Test Procedure:

- Step 1 – Authenticate and Capture All Issued Cookies

1. Log in on Device 1 using valid credentials and complete MFA.
2. Intercept the authentication response.
3. Record all cookies issued by the application or SSO domain.

> Example:
auth-cookie=XYZ123
session_meta=ABC456
tracking_id=DEF789

- Step 2 – Identify Authentication-Critical Cookie(s)

> The purpose of this step is to determine whether the SSO implementation depends on a single authentication artifact.

Reviewer: Should the text above (and below) be less "cookie"centric?

1. Send requests to protected endpoints.
2. Remove cookies individually and resend the request.
3. Observe which cookies are required to maintain a 200 OK response.

If only one cookie (e.g., auth-cookie) is required to sustain authentication, this indicates that the SSO trust boundary may rely solely on that cookie.

> Security Note:
When a single cookie represents the entire authenticated state, it becomes a high-value target. An attacker who obtains that cookie may not require any additional session metadata or device context to gain access.

- Step 3 – Cross-Device Session Injection

1. Copy the authentication-critical cookie value.
2. Open a different browser or device (Device 2).
3. Manually insert the copied single cookie.
4. Attempt to access protected endpoints across:

- The original application

> - Other applications participating in the same SSO ecosystem

Reviewer: ```suggestion
4. Attempt to access protected endpoints across:
    - The original application
    - Other applications participating in the same SSO ecosystem
```
